File: app/src/main/java/ch/bailu/aat/views/description/DescriptionLabelTextView.java
package ch.bailu.aat.views.description;

import android.content.Context;

import ch.bailu.aat.description.ContentDescription;
import ch.bailu.aat.dispatcher.OnContentUpdatedInterface;
import ch.bailu.aat.gpx.GpxInformation;
import ch.bailu.aat.util.ui.UiTheme;
import ch.bailu.aat.views.LabelTextView;

public class DescriptionLabelTextView extends LabelTextView implements OnContentUpdatedInterface {
    private final ContentDescription description;

    public DescriptionLabelTextView(Context context, ContentDescription d, UiTheme theme) {
        super(true, context, d.getLabel(), theme);

        description = d;
        setText();
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        description.onContentUpdated(iid, info);
        setText();
    }

    public void setText() {
        setText(description.getValueAsString());
    }
}

File: aat-gtk/src/main/kotlin/ch/bailu/aat_gtk/view/description/DescriptionLabelTextView.kt
package ch.bailu.aat_gtk.view.description

import ch.bailu.aat_gtk.view.LabelTextView
import ch.bailu.aat_lib.description.ContentDescription
import ch.bailu.aat_lib.dispatcher.TargetInterface
import ch.bailu.aat_lib.gpx.information.GpxInformation

class DescriptionLabelTextView(private val description: ContentDescription)
    : LabelTextView(description.getLabel()), TargetInterface {

    init {
        text = description.getValueAsString()
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        description.onContentUpdated(iid, info)
        text = description.getValueAsString()
    }
}
